app/src/main/java/org/openobservatory/ooniprobe/activity/BrowserActivity.java

```
@@ -0,0 +1,254 @@
+        switch (view.getId()) {
+            case R.id.webviewBack:
+                isWebViewCanGoBack();
+                break;
+            case R.id.webviewForward:
+                if (webView.canGoForward())
+                    webView.goForward();
+                break;
+            case R.id.webviewReload:
+                String url = webView.getUrl();
+                LoadWebViewUrl(url);
+                break;
+            case R.id.webviewClose:
+                finish();
+                break;
+            case R.id.shareButton:
+                shareUrl();
+                break;
+            case R.id.tryMirror:
+                tryMirror();
+                break;
+        }
+    }
+
+
+    private class MyWebViewClient extends WebViewClient {
+        @Override
+        public boolean shouldOverrideUrlLoading(WebView view, String url) {
+            view.loadUrl(url);
+            return true;
+
+        }
+
+        @Override
+        public void onPageStarted(WebView view, String url, Bitmap favicon) {
+            super.onPageStarted(view, url, favicon);
+            refresh.setVisibility(View.GONE);
+            urlLabel.setText(getString(R.string.loading));
+            if (!webViewProgressBar.isShown())
+                webViewProgressBar.setVisibility(View.VISIBLE);
+        }
+
+        @Override
+        public void onPageFinished(WebView view, String url) {
+            super.onPageFinished(view, url);
+            refresh.setVisibility(View.VISIBLE);
+            if (webViewProgressBar.isShown())
+                webViewProgressBar.setVisibility(View.GONE);
+            System.out.println("Finished loading " + url);
+            if (url.substring(0, 5).equals("https"))
+                urlLabel.setText("\uD83D\uDD12 " + url);
+            else
+                urlLabel.setText(url);
+            reloadButtons();
+        }
+
+        //TODO keep the error toasts for all messages? are they necessary?
```

Yes, I think so. We can perhaps improve on them, but I'd avoid ignoring errors.